src/datasets/utils/file_utils.py

```diff
@@ -327,6 +328,23 @@ def _request_with_retry(
     return response
 
 
+def fsspec_head(url, timeout=10.0):
+    _raise_if_offline_mode_is_enabled(f"Tried to reach {url}")
+    try:
+        fsspec.filesystem(urlparse(url).scheme).info(url, timeout=timeout)
+    except Exception:
+        return False
+    return True
+
+
+def fsspec_get(url, temp_file, timeout=10.0):
+    _raise_if_offline_mode_is_enabled(f"Tried to reach {url}")
+    try:
+        fsspec.filesystem(urlparse(url).scheme).get(url, temp_file, timeout=timeout)
```

Comment:
Also you may need to use `fsspec.get_fs_token_paths` first to instantiate the filesystem in case some filesystem kwargs can be parsed from the URL (it concerns all the filesystems that implement `_get_kwargs_from_urls` including gcsfs)